.github/workflows/linux_meson.yml

```diff
@@ -102,3 +102,111 @@ jobs:
+        python -m pip install matplotlib scikit-umfpack scikit-sparse
+        python -m pip uninstall -y scipy
+        python -m pip install pytest-cov coverage codecov gcovr
+
+    - name:  Prepare compiler cache
+      id:    prep-ccache
+      shell: bash
+      run: |
+        mkdir -p "${CCACHE_DIR}"
+        echo "::set-output name=dir::$CCACHE_DIR"
+        NOW=$(date -u +"%F-%T")
+        echo "::set-output name=timestamp::${NOW}"
+
+    - name: Setup compiler cache
+      uses:  actions/cache@v3
+      id:    cache-ccache
+      # Reference: https://docs.github.com/en/actions/guides/caching-dependencies-to-speed-up-workflows#matching-a-cache-key
+      # NOTE: The caching strategy is modeled in a way that it will always have a unique cache key for each workflow run
+      # (even if the same workflow is run multiple times). The restore keys are not unique and for a partial match, they will
+      # return the most recently created cache entry, according to the GitHub Action Docs.
+      with:
+        path: ${{ steps.prep-ccache.outputs.dir }}
+        # Restores ccache from either a previous build on this branch or on main
+        key:  ${{ github.workflow }}-${{ matrix.python-version }}-ccache-linux-${{ steps.prep-ccache.outputs.timestamp }}
+        # This evaluates to `Linux Tests-3.9-ccache-linux-` which is not unique. As the CI matrix is expanded, this will
+        # need to be updated to be unique so that the cache is not restored from a different job altogether.
+        restore-keys: |
+          ${{ github.workflow }}-${{ matrix.python-version }}-ccache-linux-
+
+    - name: Setup blas 64
+      run: |
+        target=$(python tools/openblas_support.py)
+        echo "[openblas64_]" > site.cfg
+        echo "library_dirs = $target/lib" >> site.cfg
+        echo "include_dirs = $target/include" >> site.cfg
+        echo "runtime_library_dirs = $target/lib" >> site.cfg
+
+    - name: Setup build and install scipy
+      run: |
+        python dev.py -j 2 --build-only --gcov
+
+    - name: Ccache performance
+      shell: bash -l {0}
+      run: ccache -s
+
+    - name: Check installation
+      run: |
+        pushd tools
+        python check_installation.py ${{ env.INSTALLDIR }}
+        popd
+
+    - name: Test SciPy
+      run: |
+        export OMP_NUM_THREADS=2
+        export OPENBLAS_NUM_THREADS=1
+        python dev.py -n -j 2 --mode full --coverage
+
+    - name: Upload coverage information
+      run: |
+        set -euo pipefail
+        RUN_DIR=`echo build-install/lib/python*/site-packages`
+        # Produce gcov output for codecov to find
+        find build -name '*.gcno' -type f -exec gcov -pb {} +
+        mv *.gcov "$RUN_DIR/"
+        # Run codecov
+        cd "$RUN_DIR"
+        curl https://codecov.io/bash -o codecov-upload.sh
+        chmod +x codecov-upload.sh
+        # Skip running gcov since we have already done it
+        ./codecov-upload.sh -X gcov
```

Comment:
fyi, the v3 version of the GitHub action now supports gcov if you wanted to try it

```
    - name: Upload coverage information
      uses: codecov/codecov-action@v3
      with:
        gcov: true

Reply:
Thank you @thomasrockhu-codecov for dropping by. I suppose @tirthasheshpatel you can have a look in a follow up since you were looking at codecov.

Reply:
Gcov GitHub action sounds great! I will take a look into it when submitting the codecov PR.

Reply:
For sure @tirthasheshpatel! Let me know if you need any help